src/components/SelectChainDrawer/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { Chain, currencyEquals } from '@pangolindex/sdk';
+import React, { memo, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { isMobile } from 'react-device-detect';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import { FixedSizeList } from 'react-window';
+import Drawer from 'src/components/Drawer';
+import usePrevious from 'src/hooks/usePrevious';
+import { Box, TextInput } from '..';
+import ChainRow from './ChainRow';
+import { filterChains } from './filtering';
+import { ChainList } from './styled';
+
+interface Props {
+  isOpen: boolean;
+  onClose: () => void;
+  onChainSelect: (chain: Chain) => void;
+  chains?: Chain[];
+  selectedChain?: Chain;
+  otherSelectedChain?: Chain;
+}
+
+const SelectChainDrawer: React.FC<Props> = (props) => {
+  const { isOpen, onClose, onChainSelect, chains, otherSelectedChain, selectedChain } = props;
+  const [searchQuery, setSearchQuery] = useState<string>('');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  const lastOpen = usePrevious(isOpen);
+
+  useEffect(() => {
+    if (isOpen && !lastOpen) {
+      setSearchQuery('');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
+
+  useEffect(() => {
+    if (isOpen && inputRef.current) {
+      setTimeout(() => {
+        inputRef.current?.focus();
+      }, 500);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
+
+  const filteredChains: Chain[] = useMemo(() => {
+    return filterChains(chains || [], searchQuery);
+  }, [chains, searchQuery]);
+
+  const filteredSortedChains: Chain[] = useMemo(() => {
+    const symbolMatch = searchQuery
+      .toLowerCase()
+      .split(/\s+/)
+      .filter((s) => s.length > 0);
+    if (symbolMatch.length > 1) return filteredChains;
+
+    return [
+      ...filteredChains.filter((chain) => chain.symbol?.toLowerCase() === symbolMatch[0]),
+      ...filteredChains.filter((chain) => chain.symbol?.toLowerCase() !== symbolMatch[0]),
+    ];
+  }, [filteredChains, searchQuery]);
```

Comment:
I took this part directly from here; 
`const symbolMatch`
It is designed for this kind of query; "ETH ETHEREUM"
`return [`
 It makes sorting basically.

But yeah, the question is, Is it necessary for the chain too? In my opinion, the answer is no. But I used it like this because it was used like that. I can refactor it if we both agree it's not necessary.